### PR TITLE
jQuery datpicker()

### PR DESCRIPTION
**Pull request by shumoo**
Replaced drop down menu from discussion with a datepicker input field. Added jquery-ui.js which should probably be thinned out or included differently.  I didn't want to delay putting this up since RJ was kind enough to help me.  I'll fix whatever needs to be updated.

### CODE REVIEW

File: class.eventcalendar.plugin.php [hunk 1]
```diff
@@ -1,265 +1,279 @@
...
+    */ 
+   public function PostController_BeforeBodyInput_Handler($Sender) {
+      if(!CheckPermission(array('Plugins.EventCalendar.Add', 'Plugins.EventCalendar.Manage'))) {
+         return;
+      }
+
+      $Sender->AddJsFile('eventcalendar.js', 'plugins/EventCalendar');
+	  $Sender->AddCssFile('eventcalendar.css', 'plugins/EventCalendar');
+	  $Sender->AddCssFile('jquery-ui.css', 'plugins/EventCalendar');	  	
+	  $Sender->AddJsFile('jquery-ui.js', 'plugins/EventCalendar');
```
Comment (R-J) [inline class.eventcalendar.plugin.php]: There is a version of jquery-ui.js shipped with Vanilla. You should use that in order to avoid version conflicts.

File: class.eventcalendar.plugin.php [hunk 2]
```diff
@@ -1,265 +1,279 @@
...
+    * @param object $Sender PostController
+    */ 
+   public function PostController_BeforeBodyInput_Handler($Sender) {
+      if(!CheckPermission(array('Plugins.EventCalendar.Add', 'Plugins.EventCalendar.Manage'))) {
+         return;
+      }
+
+      $Sender->AddJsFile('eventcalendar.js', 'plugins/EventCalendar');
+	  $Sender->AddCssFile('eventcalendar.css', 'plugins/EventCalendar');
+	  $Sender->AddCssFile('jquery-ui.css', 'plugins/EventCalendar');
```
Comment (R-J) [inline class.eventcalendar.plugin.php]: I'm not sure what the best way would be: adding the css like you did, use the css from a CDN or just add the markup needed for the calendar to the custom css.
From all that, using a CDN would be my favorite...

File: class.eventcalendar.plugin.php [hunk 3]
```diff
@@ -1,265 +1,279 @@
...
+	  $Sender->AddJsFile('jquery-ui.js', 'plugins/EventCalendar');
+      $Sender->AddDefinition('EventCalendarCategoryIDs', json_encode(C('Plugins.EventCalendar.CategoryIDs')));
+
+      // initially don't hide elements in allowed categories
+      $CategoryID = $Sender->Discussion->CategoryID;
+      if (!in_array($CategoryID, C('Plugins.EventCalendar.CategoryIDs'))) {
+         $Hidden = ' Hidden';   
+      }
+	  
+	  //Removed when switching drop down fields to a date picker
```
Comment (R-J) [inline class.eventcalendar.plugin.php]: If you manage to create the pull request so that the files are comparable, you could safely delete instead of commenting out ;)

File: class.eventcalendar.plugin.php [hunk 4]
```diff
@@ -1,265 +1,279 @@
...
+   public function DiscussionModel_BeforeSaveDiscussion_Handler($Sender) {
+      $Session = Gdn::Session();
+      $CategoryID = $Sender->EventArguments['FormPostValues']['CategoryID'];
+      // Reset event date and return if wrong category or no right to add event
+      if (!in_array($CategoryID, C('Plugins.EventCalendar.CategoryIDs')) || !$Session->CheckPermission(array('Plugins.EventCalendar.Add', 'Plugins.EventCalendar.Manage'))) {
+         $Sender->EventArguments['FormPostValues']['EventCalendarDate'] = '';
+         return;
+      }	  
+	  //formatting the date correctly since datepicker uses a different format
+	  $time = strtotime($Sender->EventArguments['FormPostValues']['EventCalendarDate']);
```
Comment (R-J) [inline class.eventcalendar.plugin.php]: You should stick to the naming conventions. So better use $Time instead of $time. Shouldn't it be better $Date anyway?

File: class.eventcalendar.plugin.php [hunk 5]
```diff
@@ -1,265 +1,279 @@
...
+      $Session = Gdn::Session();
+      $CategoryID = $Sender->EventArguments['FormPostValues']['CategoryID'];
+      // Reset event date and return if wrong category or no right to add event
+      if (!in_array($CategoryID, C('Plugins.EventCalendar.CategoryIDs')) || !$Session->CheckPermission(array('Plugins.EventCalendar.Add', 'Plugins.EventCalendar.Manage'))) {
+         $Sender->EventArguments['FormPostValues']['EventCalendarDate'] = '';
+         return;
+      }	  
+	  //formatting the date correctly since datepicker uses a different format
+	  $time = strtotime($Sender->EventArguments['FormPostValues']['EventCalendarDate']);	  
+	  $formatted_date = date('Y-m-d',$time);
```
Comment (R-J) [inline class.eventcalendar.plugin.php]: See above: $FormattedDate

File: class.eventcalendar.plugin.php [hunk 6]
```diff
@@ -1,265 +1,279 @@
...
+    * Returns nicely formatted html for an event date 
+    *
+    * @param date $EventDate Date of the event
+    * @return string html string showing the event date (translatable)
+    */ 
+   private function FormatEventCalendarDate($EventDate, $IncludeIcon = FALSE) {
+      if(!CheckPermission(array('Plugins.EventCalendar.View'))) {
+         return;
+      }
+      if ($EventDate != '0000-00-00' & !is_null($EventDate))  {
```
Comment (R-J) [inline class.eventcalendar.plugin.php]: I never officially learned PHP and that's why I'm not ashamed to ask: mustn't it be "&&" instead of "&"?
But I would prefer to change the function to `private function FormatEventCalendarDate($EventDate = '0000-00-00' , $IncludeIcon = FALSE) {` to avoid the null check

File: design/eventcalendar.css [hunk 7]
```diff
@@ -49,11 +49,11 @@
 
 li.Sunday {
    background-color: #CFECFF;
-   color: red;
+   color: #red;
```
Comment (R-J) [inline design/eventcalendar.css]: That is wrong. It must really be `color: red`

File: design/eventcalendar.css [hunk 8]
```diff
@@ -49,11 +49,11 @@
 
 li.Sunday {
    background-color: #CFECFF;
-   color: red;
+   color: #red;
 }
 li.Sunday > a {
    /* background-color: #CFECFF; */
-   color: red;
+   color: #red;
```
Comment (R-J) [inline design/eventcalendar.css]: See above.

**Comment by R-J** [comment]
I have commented the changes above. My biggest concern is that the changes in the plugin couldn't be identified. I made a diff with a text edit which was quite painful if you take into consideration that this should be an advantage of using git ;)

The files `views/month.php` and `js/eventcalendar.js` are not changed at all.

File: class.eventcalendar.plugin.php (hunk 1 above)
Comment (shumoo) [inline class.eventcalendar.plugin.php]: completed this...will check in with the rest of the updates. and do another pull request.

File: class.eventcalendar.plugin.php (hunk 3 above)
Comment (shumoo) [inline class.eventcalendar.plugin.php]: old habbits...

File: class.eventcalendar.plugin.php (hunk 4 above)
Comment (shumoo) [inline class.eventcalendar.plugin.php]: I guess so, reading up on the datatype being time made me think it would be appropriate but I can see your perspective too.

**Comment by shumoo** [comment]
I made multiple updates per R-J's comments.  The only thing that remains unchanged from the original pull is that I have not changed the new CSS file and how it's being imported; I don't know how to go about getting or creating a CDN at the moment.

**Comment by shumoo** [comment]
that was easier than I thought it might be...
